design-add-and-search-words-data-structure/nhistory.js

```diff
@@ -0,0 +1,37 @@
+var WordDictionary = function () {
+  this.dictionary = new Set();
+};
+
+/**
+ * @param {string} word
+ * @return {void}
+ */
+WordDictionary.prototype.addWord = function (word) {
+  this.dictionary.add(word);
+};
+
+/**
+ * @param {string} word
+ * @return {boolean}
+ */
+WordDictionary.prototype.search = function (word) {
+  if (word.indexOf(".") != -1) {
+    // Case of word has a '.'
+    for (let str of this.dictionary) {
+      if (str.length != word.length) continue;
+      let i;
+      for (i = 0; i < word.length; i++) {
+        if (word[i] === ".") continue;
+        if (word[i] != str[i]) break;
+      }
+      if (i === str.length) return true;
+    }
+    return false;
+  } else {
+    return this.dictionary.has(word);
+  }
+};
+
+// n: number of node | m: length of the word
```

Comment:
node가 입력으로 주어지지 않는데, 어떻게 node의 수로 복잡도가 분석될 수 있을까요? 혹시 단어의 개수를 의미하신 것은 아니신지...

```suggestion
// n: number of words | m: length of the word
```

Reply:
네 말씀하신대로 word의 숫자가 맞는데요
이걸 하나의 노드로 생각하고 적으면서 풀다가 여기서도 이렇게 적었던 것 같네요 🤦‍♂️
피드백 감사드립니다.